jax/_src/numpy/util.py

```diff
@@ -117,6 +119,44 @@ def _parse_extra_params(extra_params: str) -> dict[str, str]:
   return {p.partition(' : ')[0].partition(', ')[0]: p for p in parameters}
 
 
+def _get_device_set(x: ArrayLike | xc.Device | Sharding | None) -> Set[xc.Device] | None:
```

Comment:
I think this function should be deleted. It has no use as far as I can see and it error prone.